packages/pigeon/lib/swift_generator.dart

```diff
@@ -135,33 +137,55 @@ class SwiftGenerator extends StructuredGenerator<SwiftOptions> {
     final String readerName = '${codecName}Reader';
     final String writerName = '${codecName}Writer';
 
-    final Iterable<EnumeratedType> allTypes = getEnumeratedTypes(root);
+    final List<EnumeratedType> enumeratedTypes =
+        getEnumeratedTypes(root).toList();
+
+    void writeDecodeLogic(EnumeratedType customType) {
+      indent.writeln('case ${customType.enumeration}:');
+      indent.nest(1, () {
+        if (customType.type == CustomTypes.customEnum) {
+          indent.writeln('var enumResult: ${customType.name}? = nil');
+          indent.writeln(
+              'let enumResultAsInt: Int? = nilOrValue(self.readValue() as? Int)');
+          indent.writeScoped('if let enumResultAsInt = enumResultAsInt {', '}',
+              () {
+            indent.writeln(
+                'enumResult = ${customType.name}(rawValue: enumResultAsInt)');
```

Comment:
Optional: it doesn't look like we need `enumResult` at all; we could `return` the RHS here, and `return nil` below. One line shorter per enum, and half as many boilerplate variables.